posts/sgx.md

```diff
@@ -0,0 +1,164 @@
+   ```
+  
+TruCE server: Standalone process that registers with Intel Attestation Service and assists in remote attestation of RestAssured platform enclaves.
+
+TruCE SDK: Toolkit for application development. It has API and libraries for trusted (enclave) part of the cloud application, untrusted part of the cloud application, and the off-cloud client code that interacts with the cloud application.
+
+Trust Management Framework can run in a simulated IAS mode so that TruCE doesnt need registration with Intel. It doesnt contact the IAS and skips the attestation report signature verification step.
+
+### [RA-TLS](https://github.com/cloud-security-research/sgx-ra-tls)
+The project provides a ***proof-of-concept implementation*** on how to integrate Intel SGX remote attestation into the TLS connection setup. It extends the standard X.509 certificate with SGX-related information
+
+These are the instructions to start SGX Server.
+First of all, clone the github repository using `git clone github.com/cloud-security-research/sgx-ra-tls.git`
+Into the folder `sgx-ra-tls`, build the docker file `Dockerfile-ubuntu18.04` with parameters `-t ratls `.
+*Note:* The following command needs root priviliges to be able to use /dev/isgx
+
+To run the docker file: `sudo docker run --device=/dev/isgx --privileged=true -v /var/run/aesmd:/var/run/aesmd -v$(pwd):/project -it ratls bash`
+
+Into the folder project, run `SPID=your_spid EPID_SUBSCRIPTION_KEY=your_epid_subscription_key QUOTE_TYPE=SGX_LINKABLE_SIGNATURE bash ra_tls_options.c.sh > ra_tls_options.c`
+Finally to start the Server, `./build.sh sgxsdk ` and `( cd deps/wolfssl-examples/SGX_Linux ; ./App -s )`
+
+To run the Client use the same docker file, but previously get the container Id with docker ps.
+Using this ID, run `docker exec -ti --user root [container id] bash.`
+execute `./project/wolfssl-client . to start teh client`
+
+This screen shows the execution
+![](https://i.imgur.com/FhkJ9fn.png)
+
+### [Gramine](https://gramineproject.io/)
+Gramine is a lightweight guest OS, designed to run a single Linux application with minimal host requirements. Gramine can run applications in an isolated environment with benefits comparable to running a complete OS in a virtual machine – including guest customization, ease of porting to different host OS, and process migration.
+
+Gramine supports running Linux applications using the Intel SGX (Software Guard Extensions) technology (sometimes this version is called Gramine-SGX). With Intel SGX, applications are secured in hardware-encrypted memory regions (called SGX enclaves). SGX protects code and data in the enclave against privileged software attacks and against physical attacks on the hardware off the CPU package (e.g., cold-boot attacks on RAM). Gramine is able to run unmodified applications inside SGX enclaves, without the toll of manually porting the application to the SGX environment.
+
+## Glossary
+
+### Relay secrets through unsafe proxy server
+
```

Comment:
This is not a glossary. It looks like an appendix but I'm not sure that it's needed.